Src/StackifyLib/Utils/StackifyAPILogger.cs

```diff
@@ -27,66 +34,114 @@ public static bool LogEnabled
         {
             get
             {
-                EvaluateLogEnabled();
-                return _LogEnabled ?? false;
+                var enabled = _logEnabled ?? false;
+
+                return enabled;
             }
             set
             {
-                _LogEnabled = value;
-                Log("Logging enabled via code to value " + value.ToString());
+                _logEnabled = value;
             }
         }
-            
 
         public static StringWriter Logger
         {
             set
             {
-                _Logger = value;
+                _logger = value;
             }
         }
 
+
+
         public static void Log(string message, bool logAnyways = false)
```

Comment:
Have this call into the Log method below and handle the null ex case there. Duplicated code.